preprocess.lua

```diff
@@ -57,28 +60,48 @@ local function main()
 
   local data = { dataType=dataType }
 
+  -- keep processing options in the structure for further traceability
+  data.opt = opt
+
   data.dicts = {}
-  data.dicts.src = Vocabulary.init('source',
-                                   opt.train_src or opt.train,
-                                   opt.src_vocab or opt.vocab,
-                                   opt.src_vocab_size or opt.vocab_size,
-                                   opt.src_words_min_frequency or opt.words_min_frequency,
-                                   opt.features_vocabs_prefix,
-                                   function(s) return isValid(s, opt.src_seq_length or opt.seq_length) end)
+
+  _G.logger:info('Preparing vocabulary...')
+  if dataType ~= 'feattext' then
+    local src_file = opt.train_src
+    if dataType == 'monotext' then
+      src_file = opt.train
+    end
+    data.dicts.src = Vocabulary.init('train',
```

Comment:
Can we keep `'source'` instead? See https://github.com/OpenNMT/OpenNMT/pull/162.